### PR TITLE
Create new settleReleaseRequest method to perform common actions when creating pegouts

### PR DESCRIPTION
**Pull request by julia-zack**
-Create a new settleReleaseRequest that perform all the needed actions whenever creating a pegout.
-Get rid of existing addToPegoutsWaitingForConfirmations method.

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java [hunk 1]
```diff
@@ -1145,24 +1123,63 @@ private void processPegoutRequests(Transaction rskTx) {
...
-            pegoutWaitingForConfirmations.add(generatedTransaction, rskExecutionBlock.getNumber(), pegoutCreationRskTxHash);
-            // For a short time period, there could be items in the pegout request queue that don't have the pegoutCreationRskTxHash
-            // (these are pegouts created right before the consensus rule activation, that weren't processed before its activation)
-            // We shouldn't generate the event for those pegouts
-            if (pegoutCreationRskTxHash != null) {
-                eventLogger.logReleaseBtcRequested(pegoutCreationRskTxHash.getBytes(), generatedTransaction, amount);
-            }
-        } else {
-            pegoutWaitingForConfirmations.add(generatedTransaction, rskExecutionBlock.getNumber());
+    private void settleReleaseRequest(PegoutsWaitingForConfirmations pegoutsWaitingForConfirmations, BtcTransaction pegoutTransaction, Keccak256 releaseCreationTxHash, Coin requestedAmount) {
```
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java:1126]: I don't think we need to pass around the `pegoutsWaitingForConfirmations`, we can simply get it from the storageProvider right?

**Review by marcos-iov — COMMENTED** [review]
Looking nice and clean. Well done!

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java (hunk 1 above)
Comment (julia-zack) [reply rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java:1126]: yes, thats right, but the `getPegoutsWaitingForConfirmations` could throw an `IOException` that has to be handled all over the code. See for example in [processPegoutsInBatch](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L1205) and [processPegoutsIndividually](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L1166), that the map is an input of the method already handled in [processPegoutRequests](https://github.com/rsksmart/rskj/blob/master/rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java#L1120) method.

I thought it was better to do it this way to have less impact. But let me know what you think!

Comment (marcos-iov) [reply rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java:1126]: Yeah, no big deal. Thanks

**Review by marcos-iov — APPROVED** [review]

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java (hunk 1 above)
Comment (julia-zack) [reply rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java:1126]: thanks to you :)

**Pull request merged**